### PR TITLE
AMD GPU Monitoring

### PR DESCRIPTION
**Pull request by PurplefinNeptuna**
Adds basic AMD GPU monitoring.

Screenshot_20240415_102714

tested with RX 6700XT on Linux (ROCm 6.0.2).

working well also with nvidia gpu (library usage is skipped).

### CODE REVIEW

**Comment by PurplefinNeptuna** [comment]
image
Not working for Windows
so it's linux only

**Comment by PurplefinNeptuna** [comment]
image
Added support for amdsmi which usually already installed in system-wide python, it has temperature readings.
no pypi wheel for it, so i'm not put it to requirements.txt and only add it as an optional imports
Although amdsmi project file is available locally in `/opt/rocm/share/amd_smi/` for ROCm user.

**Comment by crystian** [comment]
I'll check in these days, thanks @PurplefinNeptuna

File: general/gpu.py [hunk 1]
```diff
@@ -27,22 +36,41 @@ def __init__(self):
...
-      self.pynvmlLoaded = False
       logger.error('Could not init pynvml.' + str(e))
 
-    if self.pynvmlLoaded and pynvml.nvmlDeviceGetCount() > 0:
-      self.cudaDevicesFound = pynvml.nvmlDeviceGetCount()
+    if not self.pynvmlLoaded:
+      try:
+        amdsmi.amdsmi_init()
+        self.amdsmiLoaded = True
+        self.needRefresh = True # bug in amdsmi
```
Comment (PurplefinNeptuna) [inline general/gpu.py:47]: Old amdsmi can't refresh it's telemetry without reinit. New version of ROCm (6.1) fix that issue, but that version has no official support for pytorch yet.

**Comment by crystian** [comment]
Sorry @PurplefinNeptuna this weekend I'm going to merge/review

**Comment by PurplefinNeptuna** [comment]
huh? gpu.py whitespace changed so much

**Comment by PurplefinNeptuna** [comment]
@crystian what will happened with gpu monitoring when comfyui use the new menu?

**Comment by crystian** [comment]
@PurplefinNeptuna right! by other PR, I'll change it
by the other menu, I'm not sure right now, I will wait to more stable version

**Pull request merged**